index.js

```diff
@@ -1 +1,151 @@
-module.exports = require('./lib/type');
+'use strict';
+/* globals Promise: false, Symbol: false */
+/* !
+ * type-detect
+ * Copyright(c) 2013 jake luer <jake@alogicalparadox.com>
+ * MIT Licensed
+ */
+var isPrototypeOfExists = typeof Object.getPrototypeOf === 'function';
+var promiseExists = typeof Promise === 'function';
+var globalObject = typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : self; // eslint-disable-line
+var isArrayExists = typeof Array.isArray === 'function';
+var symbolDoesNotExist = typeof Symbol === 'undefined';
+var toStringLeftSliceLength = 8;
+var toStringRightSliceLength = -1;
+/**
+ * ### typeOf (obj)
+ *
+ * Uses `Object.prototype.toString` to determine the type of an object,
+ * normalising behaviour across engine versions & well optimised.
+ *
+ * @param {Mixed} object
+ * @return {String} object type
+ * @api public
+ */
+module.exports = function typeDetect(obj) {
+  /* ! Speed optimisation
+   * Pre:
+   *   string literal     x 3,039,035 ops/sec ±1.62% (78 runs sampled)
+   *   boolean literal    x 1,424,138 ops/sec ±4.54% (75 runs sampled)
+   *   number literal     x 1,653,153 ops/sec ±1.91% (82 runs sampled)
+   *   null               x 8,448,576 ops/sec ±1.65% (59 runs sampled)
+   *   undefined          x 9,978,660 ops/sec ±1.92% (75 runs sampled)
+   *   function           x 2,556,769 ops/sec ±1.73% (77 runs sampled)
+   * Post:
+   *   string literal     x 38,564,796 ops/sec ±1.15% (79 runs sampled)
+   *   boolean literal    x 31,148,940 ops/sec ±1.10% (79 runs sampled)
+   *   number literal     x 32,679,330 ops/sec ±1.90% (78 runs sampled)
+   *   null               x 30,573,407 ops/sec ±0.58% (82 runs sampled)
+   *   undefined          x 32,363,368 ops/sec ±1.07% (82 runs sampled)
+   *   function           x 31,296,870 ops/sec ±0.96% (83 runs sampled)
+   */
+  var typeofObj = typeof obj;
+  if (typeofObj !== 'object') {
+    return typeofObj;
+  }
+
+  /* ! Spec Conformance
+   * ES5$15.2.4.2 (ES6$19.1.3.6) - Object.prototype.toString returns "[object Null]" for null object
+   * Test: `Object.prototype.toString.call(null)``
+   *  - PhantomJS 1.x == "[object DOMWindow]" (https://github.com/ariya/phantomjs/issues/13617)
+   *  - IE <=8  == "[object Object]"
+   */
+  if (obj === null) {
```

Comment:
I just ran tests using PhantomJS 1.9.8 and got the output below. It seems like you were right.

When it comes to DOM elements there are even some cases in which we cannot determine the specific type of the DOM Element and therefore we cannot guarantee what will be returned, for example: `Object.prototype.toString.call(document.createElement('Span'))` returns only `[object HTMLElement]` (which is too generic) instead of `[object HTMLSpanElement]`.

If we want to support Phantom1 we may start using platform specific code, so we either do this or drop support and use 'universal' code for the evergreen browsers.

**Test Output:**:
```
PhantomJS 1.9.8 (Linux 0.0.0) DOM Specific Events event FAILED
	TypeError: '[object EventConstructor]' is not a constructor (evaluating 'new Event('')')
	    at /tmp/76065da28d8ad7d390ec2964d304f3a6.browserify:356 <- test/dom.js:41:0
PhantomJS 1.9.8 (Linux 0.0.0) DOM Specific Events errorevent FAILED
	ReferenceError: Can't find variable: ErrorEvent
	    at /tmp/76065da28d8ad7d390ec2964d304f3a6.browserify:360 <- test/dom.js:45:0
PhantomJS 1.9.8 (Linux 0.0.0) DOM Specific Events hashchangeevent FAILED
	TypeError: '[object HashChangeEventConstructor]' is not a constructor (evaluating 'new HashChangeEvent('')')
	    at /tmp/76065da28d8ad7d390ec2964d304f3a6.browserify:364 <- test/dom.js:49:0
PhantomJS 1.9.8 (Linux 0.0.0) DOM Specific (HTMLElements) HTMLOptionsCollection FAILED
	AssertionError: Assertion Failed
	    at assert (/tmp/76065da28d8ad7d390ec2964d304f3a6.browserify:264 <- node_modules/simple-assert/index.js:44:0)
	    at /tmp/76065da28d8ad7d390ec2964d304f3a6.browserify:562 <- test/dom.js:247:0
PhantomJS 1.9.8 (Linux 0.0.0) DOM Specific (HTMLElements) HTMLQuoteElement FAILED
	AssertionError: Assertion Failed
	    at assert (/tmp/76065da28d8ad7d390ec2964d304f3a6.browserify:264 <- node_modules/simple-assert/index.js:44:0)
	    at /tmp/76065da28d8ad7d390ec2964d304f3a6.browserify:586 <- test/dom.js:271:0
PhantomJS 1.9.8 (Linux 0.0.0) DOM Specific (HTMLElements) HTMLSpanElement FAILED
	AssertionError: Assertion Failed
	    at assert (/tmp/76065da28d8ad7d390ec2964d304f3a6.browserify:264 <- node_modules/simple-assert/index.js:44:0)
	    at /tmp/76065da28d8ad7d390ec2964d304f3a6.browserify:599 <- test/dom.js:284:0
PhantomJS 1.9.8 (Linux 0.0.0) DOM Specific (HTMLElements) HTMLUnknownElement FAILED
	AssertionError: Assertion Failed
	    at assert (/tmp/76065da28d8ad7d390ec2964d304f3a6.browserify:264 <- node_modules/simple-assert/index.js:44:0)
	    at /tmp/76065da28d8ad7d390ec2964d304f3a6.browserify:647 <- test/dom.js:332:0
PhantomJS 1.9.8 (Linux 0.0.0): Executed 100 of 139 (7 FAILED) (skipped 39) (0.081 secs / 0.005 secs)
```